index.js

```diff
@@ -628,6 +628,50 @@ function modResponse(response, extended) {
   return response
 }
 
+function countResponseFields(resp) {
+  const counts = {};
+
+  if (resp.data) {
+    countResponseFieldsRecursive(counts, "data", resp.data);
+  }
+
+  counts.errors = resp.errors ? resp.errors.length : 0;
+  return counts;
+}
+
+function countResponseFieldsRecursive(hm, prefix, val) {
+  if (val instanceof Object === false) {
+    return;
+  }
+
+  const incr = (key, val) => {
+    if (countResponseFieldsRecursive(hm, key, val)) {
+      return;
+    }
+
+    let current = 0;
+    if (hm[key]) {
+      current = hm[key];
+    }
+    hm[key] = current + 1;
+  };
+
+  if (Array.isArray(val)) {
+    for (let i = 0; i < val.length; i++) {
+      incr(prefix, val[i]);
+    }
+
+    return true;
+  }
+
+  for (const [k, v] of Object.entries(val)) {
+    incr(`${prefix}.${k}`, v);
+  }
+
+  return false;
+}
+
+exports.countResponseFields = countResponseFields;
```

Comment:
is it intended to be used outside of the plugin?

Reply:
it is used for testing